src/tink/syntaxhub/FrontendContext.hx

```diff
@@ -100,21 +100,31 @@ class FrontendContext {
 	}
 	
 	static function moduleForType(name:String) {
-		if (name.indexOf('__impl') != -1 || plugins.getData().length == 0) return None;
+		if (name.indexOf('__impl') != -1 || plugins.getData().length == 0) return;
 		var pack = name.split('.');
-		var name = pack.pop();
-		var actual = pack.concat(['__impl', name]).join('.');
-		var exists = 
+		var tname = pack.pop();
+		var actual = pack.concat(['__impl', tname]).join('.');
+		cache[name] = {
+			pack: pack,
+			name: tname,
+			pos: Context.currentPos(),
+			fields: [],
+			kind: TDAlias(actual.asComplexType())
+		}
+		var exists =
 			try {
 				Context.getModule(actual);
 				true;
 			}
 			catch (e:Dynamic) false;
 		
 		if (!exists) {
-			var module = buildModule(pack, name);
-			if (module.types.length == 0) return None;
-			
+			var module = buildModule(pack, tname);
+			if (module.types.length == 0) {
+				cache.remove(name);
```

Comment:
This is the part that worries me the most.

While this mimics the old behavior (cleanup of modules with no types), the eventual cache removal could mean that somewhere else a consumer of `findType` has now a reference to a module that no longer exists.